### PR TITLE
CLI: Provide Typescript templates for Preact

### PR DESCRIPTION
**Pull request by yannbf**
Closes #

## What I did

The CLI only contained JS templates for Preact. This PR adds Typescript (and 4.9) to it.

## How to test

1. Run a sandbox for template, e.g. `yarn task --task sandbox --start-from auto --template preact-vite/default-ts`
2. It should contain typescript stories

## Checklist

- [ ] Make sure your changes are tested (stories and/or unit, integration, or end-to-end tests)
- [ ] Make sure to add/update documentation regarding your changes
- [ ] If you are deprecating/removing a feature, make sure to update
      [MIGRATION.MD](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md)

#### Maintainers

- [ ] If this PR should be tested against many or all sandboxes,
      make sure to add the `ci:merged` or `ci:daily` GH label to it.
- [x] Make sure this PR contains **one** of the labels below.

`["cleanup", "BREAKING CHANGE", "feature request", "bug", "documentation", "maintenance", "dependencies", "other"]`

### CODE REVIEW

File: code/renderers/preact/template/cli/ts-4-9/Header.stories.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    layout: 'fullscreen',
+  },
+} satisfies Meta<typeof Header>;
+
+export default meta;
+type Story = StoryObj<typeof meta>;
+
+export const LoggedIn: Story = {
+  args: {
+    // @ts-expect-error TODO seems like props extraction in Story type is not correct
```
Comment (yannbf) [inline code/renderers/preact/template/cli/ts-4-9/Header.stories.ts:20]: @kasperpeulen could you assist on this one? Thanks <3

File: code/renderers/preact/template/cli/ts-4-9/Header.stories.ts [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  // This component will have an automatically generated Autodocs entry: https://storybook.js.org/docs/7.0/preact/writing-docs/docs-page
+  tags: ['autodocs'],
+  parameters: {
+    // More on how to position stories at: https://storybook.js.org/docs/7.0/preact/configure/story-layout
+    layout: 'fullscreen',
+  },
+} satisfies Meta<typeof Header>;
+
+export default meta;
+type Story = StoryObj<typeof meta>;
```
Comment (kasperpeulen) [inline code/renderers/preact/template/cli/ts-4-9/Header.stories.ts]: We don't support this yet in Preact!

Comment (kasperpeulen) [reply code/renderers/preact/template/cli/ts-4-9/Header.stories.ts]: Should be `StoryObj<HeaderProps>`

**Review by kasperpeulen — CHANGES_REQUESTED** [review]
Not sure, if we should do this, maybe we should first make the types of Preact aligned with the other renderers.

**Pull request closed without merging**